core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala

```diff
@@ -0,0 +1,121 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package kafka.log
+
+import java.io.File
+import java.util.Properties
+
+import kafka.common.TopicAndPartition
+import kafka.message.ByteBufferMessageSet
+import kafka.utils.{MockTime, Pool, TestUtils}
+import org.apache.kafka.common.utils.Utils
+import org.junit.Assert._
+import org.junit.{After, Test}
+import org.scalatest.junit.JUnitSuite
+
+class LogCleanerManagerTest extends JUnitSuite {
+
+  val tmpDir = TestUtils.tempDir()
+  val logDir = TestUtils.randomPartitionLogDir(tmpDir)
+  val time = new MockTime()
+
+  @After
+  def tearDown() {
+    Utils.delete(tmpDir)
+  }
+
+  /**
+    * When checking for logs with segments ready for deletion
+    * we shouldn't consider logs where cleanup.policy=delete
+    * as they are handled by the LogManager
+    */
+  @Test
+  def testLogsWithSegmentsToDeleteShouldNotConsiderCleanupPolicyDeleteLogs() {
+    val messageSet = TestUtils.singleMessageSet("test".getBytes)
+    val log: Log = createLog(messageSet.sizeInBytes * 5, LogConfig.Delete)
+    val cleanerManager: LogCleanerManager = createCleanerManager(log)
+
+    appendMessagesAndExpireSegments(messageSet, log)
```

Comment:
In this test and the next two tests, it seems that we no longer need to append any messages to the log for testing.